File: python3/easy/0001.py
class Solution:
    def twoSum(self, nums, target):
        """(List[int], int) -> List[int] """

        seen = {}
        for position, num in enumerate(nums):
            remaining = target - num
            if remaining in seen:
                return [seen[remaining], position]
            seen[num] = position
        return []

        
solution = Solution()

print(solution.twoSum([2,3,4,11,15], 9))

File: javascript/easy/0001.js
const twoSum = (nums,target) => {
    const seen = {};

    for (let i = 0; i < nums.length; i++) {
        if (seen[nums[i]] >= 0) {
            return [seen[nums[i]], i]
        }
        seen[target - nums[i]] = i
    }
};

twoSum([2, 7, 11, 15], 9);
